dali/pipeline/data/tensor.h

```diff
@@ -40,6 +40,32 @@ class Tensor : public Buffer<Backend> {
   inline Tensor() : meta_(DALI_NHWC) {}
   inline ~Tensor() = default;
 
+  /**
+   *
+   * @brief For tensor T of shape (s_0, s_1, ..., s_{n-1}) returns a n-1 dimensional tensor T'
+   *        of shape (s_1, s_2, ..., s_{n-1}), such that
+   *        T'(x_1, x_2, ..., x_{n-1}) = T(x, x_1, x_2, ..., x_{n-1})
+   *        for param 'x' and any valid x_1, x_2, ..., x_{n-1}
+   *
+   * Returned tensor is treated as a view to this tensors and shares memory with it.
+   * @param x
+   * @return Tensor<Backend>
+   */
+  Tensor<Backend> SubspaceTensor(Index x) {
+    Tensor<Backend> view;
+    view.shape_ = std::vector<Index>(shape_.begin() + 1, shape_.end());
+    view.backend_ = backend_;
+    view.type_ = type_;
+    view.size_ = size_ / shape_[0];
+    view.num_bytes_ = view.type_.size() * view.size_;
+    // Point to data, as we are sharing use no-op deleter
+    view.data_.reset(static_cast<uint8_t *>(this->raw_mutable_data()) + x * view.num_bytes_,
+                     [](void *) {});
```

Comment:
Please correct me if I'm wrong, but wouldn't this:
```
Tensor<T> GenTesnor() { ... }

Tesnor<T>* ptr;
{
  auto t = GenTensor();
  ptr = &SubspaceTensor(1);
}
ptr->data; // here
```
be either memleak or segfault? Ofc provided, that `t` was only instance of given tensor

Reply:
Yes, that would be a segfault.
As is the case with most of the ShareData calls. The Tensor can be owning or non-owning memory, but until we change it, we have to stick with it as our API accept only Tensors for CPU ops.

Reply:
It's not my call, but I would like to express, that I strongly object committing code, that potentially introduces segfault. Cause someday, somebody will have to fix it. And at that point it might be extremely troublesome.

Reply:
Used aliasing shared_ptr, I think this should address this issue. Still should be approached during refactor and use TensorView.

Reply:
We already have a way to share tensor memory not owning it - ShareData. So it is up to developer to know what is he is doing.
Tracking that shares what is a difficult problem to make that safe and performing. I had discussion with @ptrendx about this and we decided to leave it as it is.

Reply:
@JanuszL, it will probably be the same amount of atomic operations in total if we actually share the ownership by aliasing shared pointer or not. 

Should I leave it as it is now, or revert it to separate pointer with empty deleter. The former should be safer.

Reply:
Done.